File: sample/src/main/java/com/polidea/rxandroidble3/sample/DeviceActivity.java
package com.polidea.rxandroidble3.sample;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

import com.polidea.rxandroidble3.sample.example2_connection.ConnectionExampleActivity;
import com.polidea.rxandroidble3.sample.example3_discovery.ServiceDiscoveryExampleActivity;

import butterknife.ButterKnife;
import butterknife.OnClick;

public class DeviceActivity extends AppCompatActivity {

    public static final String EXTRA_MAC_ADDRESS = "extra_mac_address";
    private String macAddress;

    @OnClick(R.id.connect)
    public void onConnectClick() {
        final Intent intent = new Intent(this, ConnectionExampleActivity.class);
        intent.putExtra(EXTRA_MAC_ADDRESS, macAddress);
        startActivity(intent);
    }

    @OnClick(R.id.discovery)
    public void onDiscoveryClick() {
        final Intent intent = new Intent(this, ServiceDiscoveryExampleActivity.class);
        intent.putExtra(EXTRA_MAC_ADDRESS, macAddress);
        startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_device);
        ButterKnife.bind(this);
        macAddress = getIntent().getStringExtra(EXTRA_MAC_ADDRESS);
        //noinspection ConstantConditions
        getSupportActionBar().setSubtitle(getString(R.string.mac_address, macAddress));
    }
}

File: sample/src/main/java/com/polidea/rxandroidble2/sample/DeviceActivity.kt
package com.polidea.rxandroidble2.sample

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import butterknife.ButterKnife
import butterknife.OnClick
import com.polidea.rxandroidble2.sample.example2_connection.newConnectionExampleActivity
import com.polidea.rxandroidble2.sample.example3_discovery.newServiceDiscoveryExampleActivity

private const val EXTRA_MAC_ADDRESS = "extra_mac_address"

internal fun Context.newDeviceActivity(macAddress: String): Intent =
    Intent(this, DeviceActivity::class.java).apply {
        putExtra(EXTRA_MAC_ADDRESS, macAddress)
    }

class DeviceActivity : AppCompatActivity() {

    private lateinit var macAddress: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_device)
        ButterKnife.bind(this)

        macAddress = intent.getStringExtra(EXTRA_MAC_ADDRESS)
        supportActionBar!!.subtitle = getString(R.string.mac_address, macAddress)
    }

    @OnClick(R.id.connect)
    fun onConnectClick() {
        startActivity(newConnectionExampleActivity(macAddress))
    }

    @OnClick(R.id.discovery)
    fun onDiscoveryClick() {
        startActivity(newServiceDiscoveryExampleActivity(macAddress))
    }
}
